controller.py

```diff
@@ -394,6 +394,53 @@ def apply_borrowing_policy(self, patron, license_pool):
             )        
         return None
 
+
+class AdminController(CirculationManagerController):
+
+    @property
+    def google(self):
+        return GoogleAuthService(
+            self._db, self.url_for('google_auth_signin'),
+            test_mode=self.manager.testing
+        )
+
+    def authenticated_admin_from_request(self):
+        authorization = flask.request.environ.get('HTTP_AUTHORIZATION')
+        if authorization:
+            admin = get_one(self._db, Admin, access_token=authorization[7:])
+            if admin and self.google.active_credentials(admin):
+                return admin
+        return self.google.auth_uri
```

Comment:
There were some type issues, but I've found a less bad-regex-y way to return the redirect from here.